agent/engine/dependencygraph/graph.go

```diff
@@ -98,16 +91,25 @@ func DependenciesAreResolved(target *api.Container, by []*api.Container) bool {
 		neededVolumeContainers[i] = volume.SourceContainer
 	}
 
-	return verifyStatusResolveable(target, nameMap, neededVolumeContainers, volumeIsResolved) &&
-		verifyStatusResolveable(target, nameMap, linksToContainerNames(target.Links), linkIsResolved) &&
-		verifyStatusResolveable(target, nameMap, target.RunDependencies, onRunIsResolved)
+	return verifyStatusResolvable(target, nameMap, neededVolumeContainers, volumeIsResolved) &&
+		verifyStatusResolvable(target, nameMap, linksToContainerNames(target.Links), linkIsResolved) &&
+		verifyStatusResolvable(target, nameMap, target.RunDependencies, onRunIsResolved)
+}
+
+func linksToContainerNames(links []string) []string {
+	names := make([]string, 0, len(links))
+	for _, link := range links {
+		name := strings.Split(link, ":")[0]
+		names = append(names, name)
+	}
+	return names
 }
 
-// verifyStatusResolveable validates that `target` can be resolved given that
+// verifyStatusResolvable validates that `target` can be resolved given that
 // target depends on `dependencies` (which are container names) and there are
 // `existingContainers` (map from name to container). The `resolves` function
 // passed should return true if the named container is resolved.
-func verifyStatusResolveable(target *api.Container, existingContainers map[string]*api.Container, dependencies []string, resolves func(*api.Container, *api.Container) bool) bool {
+func verifyStatusResolvable(target *api.Container, existingContainers map[string]*api.Container, dependencies []string, resolves func(*api.Container, *api.Container) bool) bool {
```

Comment:
No, this deals with transitions between known statuses, not about desired statuses.  The results of `DependenciesAreResolved` determine whether the container can transition to its next known status, regardless of the desired status.